langchain4j-github-models/src/main/java/dev/langchain4j/model/github/GitHubModelsChatModel.java

```diff
@@ -0,0 +1,424 @@
+    private final Long seed;
+    private final ChatCompletionsResponseFormat responseFormat;
+    private final List<ChatModelListener> listeners;
+
+    public GitHubModelsChatModel(ChatCompletionsClient client,
+                                String modelName,
+                                Integer maxTokens,
+                                Double temperature,
+                                Double topP,
+                                List<String> stop,
+                                Double presencePenalty,
+                                Double frequencyPenalty,
+                                Long seed,
+                                ChatCompletionsResponseFormat responseFormat,
+                                List<ChatModelListener> listeners) {
+
+        this(modelName, maxTokens, temperature, topP, stop, presencePenalty, frequencyPenalty, seed, responseFormat, listeners);
+        this.client = client;
+    }
+
+    public GitHubModelsChatModel(String endpoint,
+                                String serviceVersion,
+                                String gitHubToken,
+                                String modelName,
+                                Integer maxTokens,
+                                Double temperature,
+                                Double topP,
+                                List<String> stop,
+                                Double presencePenalty,
+                                Double frequencyPenalty,
+                                Long seed,
+                                ChatCompletionsResponseFormat responseFormat,
+                                Duration timeout,
+                                Integer maxRetries,
+                                ProxyOptions proxyOptions,
+                                boolean logRequestsAndResponses,
+                                List<ChatModelListener> listeners,
+                                String userAgentSuffix,
+                                Map<String, String> customHeaders) {
+
+        this(modelName, maxTokens, temperature, topP, stop, presencePenalty, frequencyPenalty, seed, responseFormat, listeners);
+        this.client = setupSyncClient(endpoint, serviceVersion, gitHubToken, timeout, maxRetries, proxyOptions, logRequestsAndResponses, userAgentSuffix, customHeaders);
+    }
+
+    private GitHubModelsChatModel(String modelName,
+                                 Integer maxTokens,
+                                 Double temperature,
+                                 Double topP,
+                                 List<String> stop,
+                                 Double presencePenalty,
+                                 Double frequencyPenalty,
+                                 Long seed,
+                                 ChatCompletionsResponseFormat responseFormat,
+                                 List<ChatModelListener> listeners) {
+
+        this.modelName = getOrDefault(modelName, DEFAULT_CHAT_MODEL_NAME);
```

Comment:
```suggestion
        this.modelName = ensureNotBlank(modelName, "modelName");
```
We are moving away from default values for model name (because they are becoming outdated pretty fast), please make it mandatory without default

Reply:
Oh I didn't know that! Let me commit the suggestion and then fix it elsewhere